### PR TITLE
remove all deprecated methods and usage

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

**Comment by balamurugana** [comment]
> Changes look fine @balamurugana are you planning to update API.md as well?

There are many PRs change APIs. Updating `API.md` wouldn't be advised for this PR.

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -4254,925 +3951,39 @@ public void removeBucket(String bucketName)
...
+   *          Object name in the bucket.
+   * @param length
+   *          Length of object data.
+   * @param data
+   *          Object data.
+   * @param uploadId
+   *          Upload ID of multipart put object.
+   * @param partNumber
+   *          Part number of multipart put object.
    */
```
Comment (kannappanr) [inline api/src/main/java/io/minio/MinioClient.java]: we should keep the comment about exceptions being thrown to be consistent

Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: Its odd to have javadoc for private methods. I would suggest to remove such unnecessary docs.

Comment (kannappanr) [reply api/src/main/java/io/minio/MinioClient.java]: ok. got it.

**Pull request merged**